components/studio/bin/hab-studio.sh

```diff
@@ -882,6 +881,12 @@ chroot_env() {
     env="$env no_proxy=$(echo "$no_proxy" | $bb sed 's/, /,/g')"
   fi
 
+  if [ -n "${SSL_CERT_FILE:-}" ] \
+  && [ -z "${NO_MOUNT}" ] \
+  && [ -z "${NO_ARTIFACT_PATH}" ]; then 
```

Comment:
Hmm, the NO_ARTIFACT_PATH test seems unnecessary.  I'd be concerned about changing an existing var that might break backward compat.  Maybe just remove the NO_ARTIFACT_PATH line here?

Reply:
I can't seem to get `-N` or setting the `NO_ARTIFACT_PATH` environment variable to actually do what it purports to do:
```
➤ hab studio --help
…
COMMON FLAGS:
…
    -N  Do not mount the source artifact cache path into the Studio (default: mount the path)
ENVIRONMENT VARIABLES:
…
    NO_ARTIFACT_PATH       If set, do not mount the source artifact cache path (`-N' flag overrides)
```
```
➤ hab studio rm
…
➤ env DEBUG=1 hab studio -N enter
…
[1][default:/src:0]# ls /hab/cache/artifacts/
core-acl-2.2.53-20190115012136-x86_64-linux.hart             core-gzip-1.9-20190115013612-x86_64-linux.hart               core-openssl-1.0.2r-20190305210149-x86_64-linux.hart
…
```
even though when I grep the debug output, I see:
```
+ NO_ARTIFACT_PATH=true
```

So, we should probably figure out whether that's a bug and if it has impact here. If it's only supposed to apply to `/hab/cache/artifacts` and will leave `/hab/cache/ssl` alone, I agree that we should just omit that check here, but I got the sense that it may have been intended to control the mounting of the entire `/hab/cache` directory and that the name was from when we didn't have other things cached, but I'm really not sure.

Reply:
This looks like a copy/paste gone wrong,  it should be `NO_CERT_PATH`, which I do think we want in this series of checks.  

@baumanj  I wasn't able to replicate the behavior you were seeing of `NO_ARTIFACT_PATH`, would you mind opening an issue on that?

Reply:
What do you see when you pass `-N` or set `NO_ARTIFACT_PATH`, @smacfarlane ?

Reply:
@baumanj If I enter an existing studio, I see an empty directory. If I destroy and create as you did, then I see the TDEPS of the studio, launcher and supervisor which is what I would expect.   `NO_ARTIFACT_CACHE` only controls the mounting of the cache from the host.  hab will still use `/hab/cache/artifacts` to cache everything it downloads, but the artifacts aren't shared with the host so will need to be fetched from builder and won't saved after the studio is removed.   Sorry I wasn't more clear this morning.